wpilibj/src/shared/java/edu/wpi/first/wpilibj/PIDController.java

```diff
@@ -284,7 +287,9 @@ protected void calculate() {
                 + calculateFeedForward();
           }
         } else {
-          if (m_I != 0) {
+          if (Math.abs(m_error) > m_maxErrorToIntegrate) {
+            m_totalError = 0.0;
+          } else if (m_I != 0) {
```

Comment:
The next iteration is going to add error anyways? I'm not sure either are more correct.